packages/lib/next-auth/auth-options.ts

```diff
@@ -90,6 +91,19 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
           }
         }
 
+        if (!user.emailVerified) {
+          const totalUserVerificationTokens = user.VerificationToken.length;
+          const lastUserVerificationToken = user.VerificationToken[totalUserVerificationTokens - 1];
```

Comment:
As per my other comment it might be nice to get the token as a dedicated item, then we can check if it's expired or was sent > X minutes ago at which point we would resend 😄